src/core/thread/key_manager.hpp

```diff
@@ -88,6 +88,38 @@ class MasterKey : public otMasterKey
 
 } OT_TOOL_PACKED_END;
 
+/**
+ * This class represents a Thread Pre-Shared Key for the Commissioner (PSKc).
+ *
+ */
+OT_TOOL_PACKED_BEGIN
+class PSKc : public otPSKc
+{
+public:
+    /**
+     * This method evaluates whether or not the Thread PSKc values match.
+     *
+     * @param[in]  aOther  The Thread PSKc to compare.
+     *
+     * @retval TRUE   If the Thread PSKc values match.
+     * @retval FALSE  If the Thread PSKc values do not match.
+     *
+     */
+    bool operator==(const PSKc &aOther) const { return memcmp(m8, aOther.m8, sizeof(PSKc)) == 0; }
+
+    /**
+     * This method evaluates whether or not the Thread PSKc values match.
+     *
+     * @param[in]  aOther  The Thread PSKc to compare.
+     *
+     * @retval TRUE   If the Thread PSKc values do not match.
+     * @retval FALSE  If the Thread PSKc values match.
+     *
+     */
+    bool operator!=(const PSKc &aOther) const { return memcmp(m8, aOther.m8, sizeof(PSKc)) != 0; }
```

Comment:
I think it would be better to implement as negative of `operator==()`:
```c++
bool operator!=(const PSKc &aOther) const { return !(*this == aOther); }
```

Reply:
I agree that this would be better way probably for more complex structures where `operator==` implementation itself may be more complex (to avoid repeating complex implementation).

But for simple structures and simple implementations like this, I am not sure if there is any advantages. Do you know of any advantages in this case?

Reply:
To me, `return !(*this == aOther)` makes clearer that `operator!=` is implemented as negative of `operator==`, but don't need to verify if the two calling of `memcmp` are the same. This avoids subtle bugs (such as wrong arguments to `memcmp`).